cmd/agent/webhook_admission.go

```diff
@@ -287,12 +290,12 @@ func setupAdmissionHandlers(ctx context.Context, platformClient *platform.Client
 		return nil, nil, nil, fmt.Errorf("start kube informer: %w", err)
 	}
 
-	apiAvailable, err := isAPIAvailable(kubeClientSet)
+	apiManagementCRDs, err := hasAPIManagementCRDs(kubeClientSet)
```

Comment:
Here `apiManagementCRDs` seems to be a weird name for a bool. The ifs below are kind of hard to get with this. I would suggest renaming the variable to `isAPIManagementAvailable` for example.

```suggestion
	isAPIManagementAvailable, err := hasAPIManagementCRDs(kubeClientSet)
```

WDYT ?